### PR TITLE
Merging Build process into master

### PR DESCRIPTION
**Pull request by Alex-Werner**
This P.R attempt to change how we are using govobject-proposal. 
More specifically it change our process. 

In the current state we were specifically provide a built version of bitcore-lib-dash, which ask us to do a maintenance on this file. 

The proposed state change this process by requiring to first build the bitcore-lib-dash before being able to use govobject-proposal (the full process in documented in the readme). 

Whenever we run a `npm run build`, a vendor/index.js file will be created, which will contains the bitcore-lib-dash dependency aswell as the code needed for the proper execution to this code. 

Nota 1 : Please wait 24hr before merging. 
Nota 2 : In the near future, specific tag will be used in package.json in order to reference which version we want. I suggest using "0.13.21".

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -15,5 +16,9 @@
   "bugs": {
     "url": "https://github.com/dashevo/govobject-proposal/issues"
   },
-  "homepage": "https://github.com/dashevo/govobject-proposal#readme"
+  "homepage": "https://github.com/dashevo/govobject-proposal#readme",
+  "devDependencies": {
+    "bitcore-lib-dash": "github:dashevo/bitcore-lib-dash",
```
Comment (nmarley) [inline package.json]: Should we specify a tag/version here so that this version of govobject-proposal doesn't break if bitcore-lib-dash master is updated?

Comment (Alex-Werner) [reply package.json]: Yes that what I mentionned on the point Nota 2, but we first need to define a tag on the github of bitcore-lib-dash, or regain access from NPM

**Comment by Alex-Werner** [comment]
Modification done : v0.13.21 reference the last version onto bitcore-lib-dash with the fix. 
From now on, when we will install the dev-dependecies it will install this known working version. 

I need a review

File: js/index.js [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+    socket = io(provider);
+
+    socket.on('connect', function() {
+        socket.emit('subscribe', 'inv');
+        console.log("socket.io initialized...");
+    });
+
+    socket.on('disconnect', function() {
+        console.log('connection lost');
+        alert('We sorry; connection lost.');
```
Comment (nmarley) [inline js/index.js:27]: :D

Comment (Alex-Werner) [reply js/index.js:27]: We want to be really sure you know the connection has been lost :p 

I understand the console.log, but an alert is maybe unnecessary in my opinion. 
I would prefer a UI things notifying that, or the button being blocked or something else

Comment (nmarley) [reply js/index.js:27]: I agree on the UI, mostly funny b/c the wording of the alert, but it would be nice to make it a UI notification.

**Comment by nmarley** [comment]
Let's get a Selenium test or at least trim the inputs before merging. If we merge this as-is right now (without input trimming), then the app will re-introduce the bug where it will allow an address with spaces, and that will then be rejected by the Dash network.

**Comment by nmarley** [comment]
Opened issue #1 for trimming input, related to above ^.

File: js/index.js (hunk 2 above)
Comment (d4046949) [reply js/index.js:27]: A nice icon on the toolbar to show connection status would be great. Also a growl to notifiy the user with a notification option similar to fb would also be of benefit

**Review by nmarley — APPROVED** [review]

**Comment by Alex-Werner** [comment]
Reproduced the fix done on the place of the conflict, which is needed before fixing the conflict issue.

**Comment by Alex-Werner** [comment]
I confirm that on a code side, it can be merged. But travis had an issue which I interpret as false alarm. 
Waiting to be confirmed first by @nmarley before merging.

**Comment by nmarley** [comment]
Yes, looks like a false alarm. Merging now, will get travis working soon.

**Pull request merged**